aspnet/client-side/bundling-and-minification.rst

```diff
@@ -1,8 +1,315 @@
+
+6.	Next, right-click the **clean** task, then select **Bindings** > **Before Build**.
+
+ 	.. image:: bundling-and-minification/_static/05-TaskRunner-BeforeBuild.png 
+
+	The **Before Build** binding option will allow the clean task to be automatically run before each time you build your project.
+
+Defining and Running a New Task
+-------------------------------
+
+To define a new gulp task, you must modify *gulpfile.js*.
+ 
+1.	Add the following JavaScript to the end of *gulpfile.js*:
+
+.. code-block:: javascript
+
+	gulp.task("first", function () {
+		console.log('first task! <-----');
+	});
+	
+	This task is named ``first`` and simply displays a string. 
+2.	Save *gulpfile.js*.
+3.	In **Solution Explorer**, right-click *gulpfile.js** and select *Task Runner Explorer*. 
+4.	In **Task Runner Explorer**, right-click **first** and select **Run**.
+
+	.. image:: bundling-and-minification/_static/06-TaskRunner-First.png 
+
+	You’ll see that the output text is displayed. If you are interested in examples based on a common scenario, see Gulp Recipes.
+
+Defining and Running Tasks in a Series
+--------------------------------------
+When you run multiple tasks, the tasks run concurrently by default. However, if you need to run tasks in a specific order, you must specify when each task is complete, as well as which tasks depend on the completion of another task. 
+
+1.	To define a series of tasks to run in order, replace the ``first`` task that you added above in *gulpfile.js* with the following:
+
+.. code-block:: javascript
+
+	gulp.task("series:first", function () {
+		console.log('first task! <-----');
+	});
+	gulp.task("series:second", ["series:first"], function () {
+		console.log('second task! <-----');
+	});
+	gulp.task("series", ["series:first", "series:second"], function () {});
+
+	You now have three tasks: ``series:first``, ``series:second``, and ``series``. The ``series:second`` task includes a second parameter that specifies an array of tasks that must be run and completed before the ``series:second`` task will run.  As specified in the code above, only the ``series:first`` task must be completed before the ``series:second`` task will run.	
+
+2.	Save *gulpfile.js*.
+3.	In **Solution Explorer**, right-click *gulpfile.js* and select **Task Runner Explorer** if it isn’t already open. 
+4.	In **Task Runner Explorer**, right-click **series** and select **Run**.
+
+	.. image:: bundling-and-minification/_static/07-TaskRunner-Series.png 
+ 
+IntelliSense
+------------
+IntelliSense, which provide coding details while you code, helps you write code faster and with fewer errors. Because the gulp tasks are created using JavaScript, you can use IntelliSense to help code. As you work with JavaScript, IntelliSense lists the objects, functions, properties, and parameters that are available based on your current context. You can select a coding option from the pop-up list provided by IntelliSense to complete the code.
```

Comment:
IntelliSense provides code completion, parameter info and other features to help you author code more productively (link to https://msdn.microsoft.com/en-us/library/hcw1s69b.aspx). Gulp tasks are written in JavaScript, therefore intellisense can ..